javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll

```diff
@@ -106,7 +106,11 @@ module DataFlow {
 
     /** Holds if this node may evaluate to the Boolean value `b`. */
     predicate mayHaveBooleanValue(boolean b) {
-      b = analyze().getAValue().(AbstractBoolean).getBooleanValue()
+      getAPredecessor().mayHaveBooleanValue(b) // needed stage 31 + stage 26 + stage 22 (unfixable)
```

Comment:
The comment here is a bit obscure and will inevitably become outdated. I'd just remove it.

Reply:
yes. That was a leftover from my development that I forgot to remove.